File: typescript/functions/string_manipulation.ts
const text: string = "  Hello, World!  ";

console.log(`Original: '${text}'`);
console.log(`Upper: '${text.toUpperCase()}'`);
console.log(`Lower: '${text.toLowerCase()}'`);
console.log(`Trim: '${text.trim()}'`);
console.log(`Replace: '${text.replace('World', 'TypeScript')}'`);
console.log(`Split: ${text.trim().split(', ')}`);
console.log(`StartsWith: ${text.trim().startsWith('Hello')}`);
console.log(`EndsWith: ${text.trim().endsWith('!')}`);
console.log(`Includes: ${text.includes('World')}`);
console.log(`Substring: '${text.trim().substring(0, 5)}'`);
console.log(`Repeat: '${'-'.repeat(20)}'`);

File: rust/functions/string_manipulation.rs
fn main() {
    let text = "  Hello, World!  ";

    println!("Original: '{}'", text);
    println!("Upper: '{}'", text.to_uppercase());
    println!("Lower: '{}'", text.to_lowercase());
    println!("Trim: '{}'", text.trim());
    println!("Replace: '{}'", text.replace("World", "Rust"));
    println!("Split: {:?}", text.trim().split(", ").collect::<Vec<&str>>());
    println!("Contains: {}", text.contains("World"));
    println!("Starts with: {}", text.trim().starts_with("Hello"));
    println!("Ends with: {}", text.trim().ends_with("!"));
    println!("Length: {}", text.trim().len());
    println!("Repeat: '{}'", "-".repeat(20));
}
